manifests/plugin/table.pp

```diff
@@ -0,0 +1,24 @@
+# https://collectd.org/wiki/index.php/Chains
+class collectd::plugin::table (
+  Enum['present', 'absent'] $ensure    = 'present',
+  Integer $order                       = 10,
+  Hash[String, Struct[{
+    'plugin'    => Optional[String],
+    'separator' => Optional[String],
+    'instance'  => Optional[String],
+    'results'   => Array[Struct[{
+      'type'            => String,
+      'values_from'     => Array[Integer, 1],
+      'instance_prefix' => Optional[String],
+      'instances_from'  => Optional[Array[Integer, 1]]
+    }], 1]
+  }], 1] $tables,
+) {
+  include ::collectd
```

Comment:
please don't add the leading :: infront of a class name. This isn't needed anymore since we don't support puppet 3.

Reply:
Done, old habbit.